blocks/init/src/blocks/components/paragraph/components/paragraph-options.js

```diff
@@ -11,45 +11,41 @@ export const paragraphSizes = [
-                  {__('Paragraph Color', 'eightshift-boilerplate')}
-                </Fragment>
-              }
-              help={__('Change Paragraph color.', 'eightshift-boilerplate')}
-              value={styleColor}
-              onChange={onChangeStyleColor}
-            />
-          }
 
-          {onChangeStyleSize &&
-            <SelectControl
-              label={__('Paragraph Font Size', 'eightshift-boilerplate')}
-              value={styleSize}
-              options={paragraphSizes}
-              onChange={onChangeStyleSize}
-            />
+      {onChangeStyleColor &&
```

Comment:
I still didn't get my answer on why do we need `onChangeStyleColor` check here? I'd really love somebody explaining that to me 🙂

Reply:
that way you can render only relevant options when using paragraph inside another component.

Say you have a hero, using Header, Paragraph, Image & Button.

You only want the user to be able to change the Header / Paragraph's color but not size or anything else. Then you could just pass `onChangeStyleColor` to `paragraph-options.js` and only that option would be exposed to the user.

Hope that makes more sense.

Reply:
I have answered this in some PR before. But never mind. We are wrapping each option component inside a conditional because if you want for example to use paragraph option in you block and you only want to use Color and not the whole set of options you can just provide a prop for color and you have only colors. If this wan't in the conditions all other props will throw an error. So this is the way we have managed to make components as versatile as possible.

Why are we wrapping a condition with and onChange method. Because of a bool value. If you wrap a bool in a condition when it is set to false it will not show at all. So for consistency sake we have agreed to wrap everything inside a onChange method rather than value. 

In general inside a blockOption you don't need to have to have conditional around the options but in components you do. So also as a good practice an consistency we have added the conditionals inside a blocks also.

Reply:
Cool, makes sense. Thanks for the explanation!